### ISSUE TITLE
Mobile app string translation

### ISSUE THREAD

**Issue by mattpen** [phetsims/rosetta#211]
We need to devise a way to translate the strings used in the mobile apps via rosetta.  There will be three major steps to this process.  

  1. Retrieve strings that need translation
  2. Add the strings to babel
  3. Package the strings for the apps.

In addition, I believe @jbphet mentioned that we should modify babel so that its very clear these strings are for apps and not sims.

[Strings format for iOS](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/LoadingResources/Strings/Strings.html)
[Strings xml spec for Android](https://developer.android.com/guide/topics/resources/string-resource)

Issue created by request from @oliver-phet.  @ariel-phet should we schedule some design time with ON, JB (or whoever the current rosetta lead is), and myself?

**Comment by mattpen** [comment]
Here is the current Android strings file: https://github.com/phetsims/phet-android-app/blob/master/app/src/main/res/values/strings.xml

**Comment by mattpen** [comment]
Here is the current iOS strings file.

The xliff format is documented here: http://docs.oasis-open.org/xliff/v1.2/os/xliff-core.html

[en.xliff.gz](https://github.com/phetsims/rosetta/files/3207934/en.xliff.gz)

**Comment by mattpen** [comment]
We should also localize the listings for the (app store)[https://github.com/phetsims/phet-ios-app/issues/483] and the (play store)[https://support.google.com/googleplay/android-developer/answer/113469].

**Comment by oliver-phet** [comment]
Design meeting 5/23:

1. App strings will be written directly to xml files from Rosetta (1 file per locale)
2. Store description will be written as a separate file from Rosetta (1 file per locale)
3. Both files stored separately in the 'phet-ios-app' and 'phet-android-app' repos
4. https://phet.colorado.edu/translate/ will be edited to add an 'Apps' section that allows a translator to select "iOS App 1.3" and a locale

QA will check Chinese (zh_TW, zh_CN), Portuguese, Spanish (es, es_MX)

**Comment by ariel-phet** [comment]
@kathy-phet wrote

> Ariel, Matt, Oliver,
> 
> I was chatting with JB today to catch up on a few things, and he let me know that the work on Rosetta was going to take quite a few hours - much more than I had anticipated .. he said easily more than 40 hours on his part which translates to more than a month -- likely quite a bit more. He said if Rosetta was perfectly extensible and high quality code it could take 40 hours, so I think the actual time could be 80+ hours with the way the code is now.
> 
> Given that situation, I would like to take a different tack on this first internationalized version - to use a more manual spreadsheet approach.
> 
> So creating a google sheet or excel sheet with the baseline strings for English, ask the translators to add the translation into the column for their language, and then when complete writing that to a csv or whatever format Matt can work with.
> 
> For the 5-10 sets of languages we are starting with, and the cost in both time and $$ to build this into Rosetta if JB does it, I would like to proceed with manual for now so we are not held up and do not have to make that level of investment into Rosetta.
> 
> JB is finishing up fixing the current bugs in Rosetta.
> 
> Best,
> Kathy

So this work is on-hold current

**Comment by ariel-phet** [comment]
I think we will still need a way to store the strings in the app repos in the correct format once we have files from translators, but perhaps that should be a separate issue.

**Comment by oliver-phet** [comment]
@mattpen It should be easy enough for me to create a spreadsheet for translators to add strings, for each "source" string. Do the "note" strings also need to be translated? 

I also wonder if it would help to send each translator a copy of the .xliff file (with non-translatable content stripped out) where they can translate the strings directly?

**Comment by mattpen** [comment]
@oliver-phet 

> Do the "note" strings also need to be translated?

No, they do not.  These are comments for developers or machine-reading only and will not appear to the end user.

<hr />

> I also wonder if it would help to send each translator a copy of the .xliff file (with non-translatable content stripped out) where they can translate the strings directly?

This would be the easiest way from my perspective if it is readable by the translators.  That way I can just commit and push the files directly to the app repo.

If we send out a spreadsheet I would need to write a script to translate from csv to xliff, and it sounds like we want to avoid costs like that.  If it for some reason must be in spreadsheet format, it would be ideal if you could include the `id` attribute of the `<trans-unit>` objects and the `original` attribute of the `<file>` object for each source string.  Something like this:

```
file,id,source,target
Phet/Base.lproj/Main.storyboard,3a9-m2-eam.headerTitle,"Simulations made with love by","Simulaciones hechas con amor por"
Phet/Base.lproj/Main.storyboard,05j-hV-VHw.title,"Filters","Filtros"
```

**Comment by oliver-phet** [comment]
@mattpen for strings like this:
`<source>Andrew, Ellie, Eric, Luis, Sheefali, Chris, the PhET team for CU Boulder's Computer Science Capstone class 2016
and
Chris Huxtable of BitFlip Labs</source>`
Do we need to preserve the line breaks?

**Comment by mattpen** [comment]
> Do we need to preserve the line breaks?

Yes, that is probably ideal.

**Comment by mattpen** [comment]
@oliver-phet and I discussed further in a zoom call.  We decided that we will just send the xliff files directly to translators along with instructions on how to translate them using [tranzapp](http://www.beyondf.com/tranzapp/translate/).  When we get the file back, we'll need to update the language attributes and filenames, then they can be loaded into the app.

@oliver-phet is also going to add a new `<file>` object in the xliff file that contains the app store listing strings that we would like to be translated.  This will allow the translator to translate everything all in one place.  When we get the file back, we'll need to extract this `<file>` object and manually copy it into the app store.

**Comment by oliver-phet** [comment]
Our procedure for collecting translations is working for now, removing assignment.

**Comment by mattpen** [comment]
XLIFF translation working for now. Removing assignment.
